cmd/deployIntegrationArtifact.go

```diff
@@ -78,7 +78,11 @@ func runDeployIntegrationArtifact(config *deployIntegrationArtifactOptions, tele
 		return errors.Wrapf(httpErr, "HTTP %v request to %v failed with error", httpMethod, deployURL)
 	}
 
-	defer deployResp.Body.Close()
+	defer func() {
+		if deployResp != nil && deployResp.Body != nil {
+			deployResp.Body.Close()
+		}
+	}()
```

Comment:
The defer should be before the error check above. Also, it is better to defer inside the if-statement [as shown here](https://github.com/SAP/jenkins-library/blob/master/cmd/malwareExecuteScan.go#L119-L121).